spec/features/user_index_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec.describe 'User Index Page', type: :feature do
+  before do
+    User.create(
+      name: 'Mehdi Ahmidou',
+      id: 1,
+      post_counter: 2,
+      photo: 'https://source.unsplash.com/featured/300x203'
+    )
+
+    User.create(
+      name: 'John deo',
+      id: 2,
+      post_counter: 5,
+      photo: 'https://source.unsplash.com/featured/300x203'
+    )
+  end
+
+  it 'displays usernames of all other users' do
+    visit users_path
+    expect(page).to have_content('Mehdi Ahmidou')
+    expect(page).to have_content('John deo')
+  end
+
+  it 'displays profile picture for each user' do
+    visit users_path
+    expect(page).to have_css("img[src*='https://source.unsplash.com/featured/300x203']", count: 2)
+  end
+
+  it 'redirects to user show page when clicking on a user' do
+    visit users_path
+    click_link('John deo')
+    expect(current_path).to eq(user_path(User.last))
+  end
+end
```

Comment:
- [ ] It seems you haven't included the below test in this file:
- Tests that assert you can see the number of posts each user has written.